frontend/src/components/Registration/Registration.tsx

```diff
@@ -1,46 +1,165 @@
-import React from 'react';
 import { FormControl, TextField, Button, Typography, Box } from '@mui/material';
 import {
   BorderForm,
   RegistrationFormWrapper,
-  StyledSpan,
   WrapButtonAndText
 } from 'components/Registration/styles';
 import { WrapH1 } from 'components/ForgotPassword/styles';
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import { useNavigate } from 'react-router-dom';
 
 function Registration() {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [emailDirty, setEmailDirty] = useState(false);
+  const [passwordDirty, setPasswordDirty] = useState(false);
+  const [emailError, setEmailError] = useState("Email can't be empty");
+  const [passwordError, setPasswordError] = useState("Password can't be empty");
+  const [formValid, setFormValid] = useState(false);
+  const [errorReq, setErrorReq] = useState('');
+  const [succesReq, setSuccesReq] = useState('');
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (emailError || passwordError) {
+      setFormValid(false);
+    } else {
+      setFormValid(true);
+    }
+  }, [emailError, passwordError]);
+
+  const registration = async () => {
+    try {
+      const response = await axios.post('http://localhost:3001/api/signup', {
+        email,
+        password
+      });
+      if (response.status === 200) {
+        setSuccesReq('Registration successful');
+        setTimeout(() => navigate('/login'), 3000);
+      }
+    } catch (e: any) {
+      setErrorReq('This email is already exists');
+      setTimeout(() => setErrorReq(''), 3000);
+    }
+  };
+
+  const emailHandler = (e: any) => {
+    setEmail(e.target.value);
+    const re =
+      /^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/i;
+    if (!re.test(String(e.target.value).toLowerCase())) {
+      setEmailError('Incorrect email');
+    } else {
+      setEmailError('');
+    }
+  };
+  const passwordHandler = (e: any) => {
+    setPassword(e.target.value);
+    if (e.target.value.length < 2 || e.target.value.length > 10) {
+      setPasswordError('Password should more 2 and less then 10 symbols');
+      if (!e.target.value) {
+        setPasswordError("Password can't be empty");
+      }
+    } else {
+      setPasswordError('');
+    }
+  };
+
+  const blurHandler = (e: any) => {
+    if (e.target.name === 'email') {
+      setEmailDirty(true);
+    }
+    if (e.target.name === 'password') {
+      setPasswordDirty(true);
+    }
+  };
   return (
     <RegistrationFormWrapper>
       <BorderForm>
         <FormControl sx={{ width: '35ch' }}>
           <WrapH1>
             <Typography sx={{ fontSize: '24px' }}> Create profile</Typography>
           </WrapH1>
-
+          {errorReq && (
+            <Box
+              style={{
+                color: 'red',

```

Comment:
I think it's bad, to use an inline color. use theme.pallete.error.main
theme u can get from useTheme hook